cli/ts/index.ts

```diff
@@ -357,6 +358,37 @@ program
       program.error((error as Error).message, { exitCode: 1 });
     }
   });
+program
+  .command("extractVk")
+  .description("extract vkey to json file")
+  .requiredOption(
+    "-pqv, --process-messages-zkey-qv <processMessagesZkeyPathQv>",
+    "the process messages qv zkey path (see different options for zkey files to use specific circuits https://maci.pse.dev/docs/trusted-setup, https://maci.pse.dev/docs/testing/#pre-compiled-artifacts-for-testing)",
+  )
+  .requiredOption(
+    "-tqv, --tally-votes-zkey-qv <tallyVotesZkeyPathQv>",
+    "the tally votes qv zkey path (see different options for zkey files to use specific circuits https://maci.pse.dev/docs/trusted-setup, https://maci.pse.dev/docs/testing/#pre-compiled-artifacts-for-testing)",
+  )
+  .requiredOption(
+    "-pnqv, --process-messages-zkey-non-qv <processMessagesZkeyPathNonQv>",
+    "the process messages non-qv zkey path (see different options for zkey files to use specific circuits https://maci.pse.dev/docs/trusted-setup, https://maci.pse.dev/docs/testing/#pre-compiled-artifacts-for-testing)",
+  )
+  .requiredOption(
+    "-tnqv, --tally-votes-zkey-non-qv <tallyVotesZkeyPathNonQv>",
+    "the tally votes non-qv zkey path (see different options for zkey files to use specific circuits https://maci.pse.dev/docs/trusted-setup, https://maci.pse.dev/docs/testing/#pre-compiled-artifacts-for-testing)",
+  )
+  .action(async (cmdObj) => {
+    try {
+      const processVkQv = await extractVk(cmdObj.processMessagesZkeyQv);
+      const tallyVkQv = await extractVk(cmdObj.tallyVotesZkeyQv);
+      const processVkNonQv = await extractVk(cmdObj.processMessagesZkeyNonQv);
+      const tallyVkNonQv = await extractVk(cmdObj.tallyVotesZkeyNonQv);
+
+      fs.writeFileSync("vks.json", JSON.stringify({ processVkQv, tallyVkQv, processVkNonQv, tallyVkNonQv }));
```

Comment:
let's use promise version

Reply:
This is not fixed

Reply:
oh you refer to fs.promises? my bad for marking this as resolved

Reply:
I also don't get it OAO my bad